cardano-testnet/src/Testnet/Runtime.hs

```diff
@@ -287,3 +289,36 @@ createSubdirectoryIfMissingNew :: ()
 createSubdirectoryIfMissingNew parent subdirectory = GHC.withFrozenCallStack $ do
   IO.createDirectoryIfMissing True $ parent </> subdirectory
   pure subdirectory
+
+-- | Start ledger state logging for the first node in the background.
+-- Logs will be placed in <tmp workspace directory>/logs/ledger-state.log
+-- The logging thread will be cancelled when `MonadResource` releases all resources.
+startLedgerStateLogging
+  :: forall m. MonadCatch m
+  => MonadResource m
+  => MonadTest m
+  => TestnetRuntime
+  -> FilePath -- ^ tmp workspace directory
+  -> m ()
+startLedgerStateLogging testnetRuntime tmpWorkspace = do
+  socketPath <- H.noteM $ H.sprocketSystemName <$> H.headM (poolSprockets testnetRuntime)
+  let logFile = makeLogDir (TmpAbsolutePath tmpWorkspace) </> "ledger-state.log"
+  _ <- runInBackground . runExceptT $
+    foldBlocks
+      (File $ configurationFile testnetRuntime)
+      (Api.File socketPath)
+      Api.QuickValidation
+      ()
+      (handler logFile)
+  pure ()
+  where
+    -- handler :: FilePath -> Env -> LedgerState -> [LedgerEvent] -> BlockInMode -> () -> IO ((), FoldStatus)
+    handler outputFp _ ledgerState _ _ _ = do
```

Comment:
If we want this additional debugging, then we ought to use the `LedgerState era` defined in `cardano-ledger`. It will be a worse experience to use the consensus definition.

Reply:
I would like to refactor this later to dump `NewEpochState`, but to do so I need to change cardano-api to expose conversion functionality.

Moreover we haven't decided how `checkLedgerStateCondition` should look like. Ideally I'd like to use it here and just print `NewEpochState` - I haven't figured out how to do it elegantly yet.